### PR TITLE
Feature/30112020 clocks

### PR DESCRIPTION
**Pull request by jorgedclopes**

### CODE REVIEW

File: src/game.py [hunk 1]
```diff
@@ -1,5 +1,13 @@
+import warnings
+from datetime import datetime
 import pgn
 
+BASE_TIME = datetime(1900, 1, 1)
+
+
+def compute_delta_time(time):
+    return (time - BASE_TIME).total_seconds()
```
Comment (duartenina) [inline src/game.py]: Why not set BASE_TIME on the datetime of the first move?

File: src/get_games.py [hunk 2]
```diff
@@ -44,16 +44,22 @@ def save_to_file(game,
 
     """
     filename = os.path.join(save_dir, save_file + '.pgn')
-    with open(filename, 'w') as file:
-        file.write("".join(game))
+    if os.path.exists(filename):
```
Comment (duartenina) [inline src/get_games.py]: Append mode already creates a file if it does not exists.

File: src/game.py [hunk 3]
```diff
@@ -126,17 +142,30 @@ def load_from_json(json_dict, source='lichess'):
...
 
         if result is None:
             raise RuntimeError("Get_result does not fall in any of the conditions.")
         return result
+
+    def get_clocks(self):
+        if isinstance(self.moves, list):
+            if "clk" in self.moves[1]:
+                clocks_strings = list(map(lambda s: s[8:15], self.moves[1::2]))
+                return list(map(
```
Comment (duartenina) [inline src/game.py]: Maybe list comprehension?

`[compute_delta_time(datetime.strptime(s, '%H:%M:%S')) for s in clocks_strings]`

File: src/game.py (hunk 1 above)
Comment (jorgedclopes) [reply src/game.py]: Sorted

File: src/get_games.py (hunk 2 above)
Comment (jorgedclopes) [reply src/get_games.py]: Sorted!

File: src/game.py (hunk 3 above)
Comment (jorgedclopes) [reply src/game.py]: Sorted!

File: src/get_games.py [hunk 4]
```diff
@@ -40,44 +40,50 @@ def save_to_file(game,
...
 def download_games(name: str,
                    db_dir: str = 'resources',
-                   pref_type: str = None,
-                   initial_time: int = None,
-                   latest_time: int = None,
+                   perf_type: str = None,
+                   time_period: list = None,
                    is_rated: bool = True,
+                   token: str = None,
+                   mock: bool = False
```
Comment (duartenina) [inline src/get_games.py]: This shouldn't be here, should it?

**Review by duartenina — COMMENTED** [review]
There's a mock argument in one of the functions.

File: src/get_games.py (hunk 4 above)
Comment (jorgedclopes) [reply src/get_games.py]: No it shouldn't.
Sorted!

File: src/get_games.py [hunk 5]
```diff
@@ -40,38 +40,38 @@ def save_to_file(game,
...
+    with open(filename, 'a') as file:
         file.write("".join(game))
 
 
 def download_games(name: str,
                    db_dir: str = 'resources',
-                   pref_type: str = None,
-                   initial_time: int = None,
-                   latest_time: int = None,
+                   perf_type: str = None,
```
Comment (targzip-dev) [inline src/get_games.py:54]: This variable name does not match the documentation below.

File: src/get_games.py [hunk 6]
```diff
@@ -144,5 +147,16 @@ def download_games(name: str,
...
+    user_stats = lichess.api.user('carequinha')
+    time_creation = user_stats['createdAt']
+    time_last_seen = user_stats['seenAt']
+    inc = int(time_last_seen - time_creation / 100)
+    time_mock = time_creation + inc
+
+    # this is a mock to demo how to use this function, the timeframes are very small
     download_games('carequinha',
-                        is_rated=True)
+                   perf_type="blitz",
```
Comment (targzip-dev) [inline src/get_games.py:152]: Check for methods that use "pref_type" or "preftype"  and uniformise usage. What does this mean, by the way?

Comment (jorgedclopes) [reply src/get_games.py:152]: Sorted. Typo in the docs. There is a game time format called blitz usually between 3 and 10 min games.

File: src/get_games.py (hunk 5 above)
Comment (jorgedclopes) [reply src/get_games.py:54]: Documentation was wrong.

File: src/get_games.py [hunk 7]
```diff
@@ -88,13 +86,14 @@ def download_games(name: str,
...
-    token = setup()
-
     # print(user['perfs'].values())
     # for i in user['perfs']:
     #    print(i, "\t", user['perfs'][i])
 
     # time in milliseconds since Jan 1st 1970
+    if time_period is None:
+        time_period = []
+    initial_time, latest_time = time_period
```
Comment (duartenina) [inline src/get_games.py:90]: Errors out if time_period was originally None.

File: src/get_games.py [hunk 8]
```diff
@@ -108,25 +107,27 @@ def download_games(name: str,
...
 
     time_30min = 30 * 60 * 1000
     increment = max([int(delta_time / 100), time_30min])
-    latest_time = initial_time + 2 * increment
+
     for time_index in range(initial_time,
                             latest_time,
                             increment):
-        print(convert_ms_to_date(time_index))
+        print(convert_ms_to_date(time_index).__str__() +
```
Comment (duartenina) [inline src/get_games.py]: 1. never call ```var.__str__ ```, call ```str(val)```
2. print() already calls str() on its arguments
3. if you want to align text, it's probably better to use f-strings

File: src/get_games.py [hunk 9]
```diff
@@ -144,5 +145,16 @@ def download_games(name: str,
 
 
 if __name__ == '__main__':  # pragma: no cover
+    auth = setup(path="./")
+
+    user_stats = lichess.api.user('carequinha')
```
Comment (duartenina) [inline src/get_games.py]: Maybe add a variable user, so you can change it more easily (see also line 157).

File: src/get_games.py (hunk 8 above)
Comment (jorgedclopes) [reply src/get_games.py]: 1) Fixed.
2) Fixed.
3) No alignment needed, just want to separate the values. Let's keep this simple for now.

File: src/get_games.py (hunk 9 above)
Comment (jorgedclopes) [reply src/get_games.py]: Fixed.

File: src/get_games.py (hunk 7 above)
Comment (jorgedclopes) [reply src/get_games.py:90]: Is this better or do you think a different pattern would be better?
I don't really like how these ifs are but...

Comment (jorgedclopes) [reply src/get_games.py:90]: Using tuples simplify a bit the logic here.

**Review by duartenina — APPROVED** [review]

**Pull request merged**